test/extended/templates/templateservicebroker_e2e.go

```diff
@@ -259,25 +280,78 @@ var _ = g.Describe("[Conformance][templates] templateservicebroker end-to-end te
 
 	deprovision := func() {
 		g.By("deprovisioning a service")
-		err := brokercli.Deprovision(context.Background(), cliUser, instanceID)
+		err := cli.TemplateClient().Template().Templates(cli.Namespace()).Delete(privatetemplate.Name, &metav1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = brokercli.Deprovision(context.Background(), cliUser, instanceID)
```

Comment:
doesn't this need to wait until lastoperation returns success?  (should have a test for that anyway)

Reply:
brokercli (already) handles that internally.  What additional test are you looking for?

Reply:
ok, I didn't dig into brokercli explicitly.  I see now that it already handles both sync and async responses (I was assuming it only handled sync and would need changes to add support for async)